Reduce the nproc when building FA on 8.9

huydhn opened the pull request:
The newly introduce sm89 build is failing consistently in trunk now because of OOM https://github.com/pytorch/pytorch/actions/runs/12186328178/job/33994606556.  I suspect that FlashAttention is the cause.

### Testing

CI to see if the build works.

huydhn committed:
Reduce the nproc when building FA on 8.9

malfet reviewed (approved):
If it works, that SGTM, but something change in our infra to cause some many OOMs recently...

malfet commented on .ci/pytorch/build.sh:
```diff
@@ -191,7 +191,7 @@ fi
 # We only build FlashAttention files for CUDA 8.0+, and they require large amounts of
 # memory to build and will OOM
-if [[ "$BUILD_ENVIRONMENT" == *cuda* ]] && [[ "$TORCH_CUDA_ARCH_LIST" == *"8.6"* || "$TORCH_CUDA_ARCH_LIST" == *"8.0"* ]]; then
+if [[ "$BUILD_ENVIRONMENT" == *cuda* ]] && [[ "$TORCH_CUDA_ARCH_LIST" > "8.0" || "$TORCH_CUDA_ARCH_LIST" == "8.0" ]]; then
```
```suggestion
if [[ "$BUILD_ENVIRONMENT" == *cuda* ]] && [[ "$TORCH_CUDA_ARCH_LIST" >= "8.0" ]]; then
```

malfet committed:
Update .ci/pytorch/build.sh

Should fix lint and will also make it a bit better

huydhn commented on .ci/pytorch/build.sh:
```diff
@@ -191,7 +191,7 @@ fi
 # We only build FlashAttention files for CUDA 8.0+, and they require large amounts of
 # memory to build and will OOM
-if [[ "$BUILD_ENVIRONMENT" == *cuda* ]] && [[ "$TORCH_CUDA_ARCH_LIST" > "8.0" || "$TORCH_CUDA_ARCH_LIST" == "8.0" ]]; then
+if [[ "$BUILD_ENVIRONMENT" == *cuda* ]] && [[ "$TORCH_CUDA_ARCH_LIST" >= "8.0" ]]; then
```
Somehow, `>=` is a wrong syntax when I test it out locally, thus the weird use of `>` and `==`.  Let's see

huydhn replied:
Yeah, bash doesn't like this syntax and shows `syntax error in conditional expression`.  Let me fix this

malfet replied:
Let's see if `-ge` will do the trick

malfet commented on .ci/pytorch/build.sh:
```diff
@@ -191,7 +191,7 @@ fi
 # We only build FlashAttention files for CUDA 8.0+, and they require large amounts of
 # memory to build and will OOM
-if [[ "$BUILD_ENVIRONMENT" == *cuda* ]] && [[ "$TORCH_CUDA_ARCH_LIST" == *"8.6"* || "$TORCH_CUDA_ARCH_LIST" == *"8.0"* ]]; then
+if [[ "$BUILD_ENVIRONMENT" == *cuda* ]] && [[ "$TORCH_CUDA_ARCH_LIST" >= "8.0" ]]; then
```
```suggestion
if [[ "$BUILD_ENVIRONMENT" == *cuda* ]] && [[ "$TORCH_CUDA_ARCH_LIST" -ge "8.0" ]]; then
```

malfet committed:
Update .ci/pytorch/build.sh

huydhn committed:
Update build.sh

https://stackoverflow.com/questions/8654051/how-can-i-compare-two-floating-point-numbers-in-bash

huydhn commented:
@pytorchbot merge -f 'Lint and sm89 jobs looks green'
